binding/web/src/orca_worker.ts

```diff
@@ -0,0 +1,303 @@
+/*
+  Copyright 2024 Picovoice Inc.
+
+  You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+  file accompanying this source.
+
+  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+  specific language governing permissions and limitations under the License.
+*/
+
+import PvWorker from 'web-worker:./orca_worker_handler.ts';
+
+import {
+  OrcaModel,
+  OrcaOptions,
+  OrcaSpeech,
+  OrcaWorkerInitResponse,
+  OrcaWorkerSynthesizeResponse,
+  OrcaWorkerReleaseResponse,
+  PvStatus,
+} from './types';
+import { loadModel } from '@picovoice/web-utils';
+
+import { pvStatusToException } from './orca_errors';
+
+export class OrcaWorker {
+  private readonly _worker: Worker;
+  private readonly _version: string;
+  private readonly _sampleRate: number;
+  private readonly _maxCharacterLimit: number;
+  private readonly _validPunctuationSymbols: string[];
+  private readonly _numSymbols: number;
+
+  private static _wasm: string;
+  private static _wasmSimd: string;
+  private static _sdk: string = 'web';
+
+  private constructor(
+    worker: Worker,
+    version: string,
+    sampleRate: number,
+    maxCharacterLimit: number,
+    validPunctuationSymbols: string[],
+    numSymbols: number,
+  ) {
+    this._worker = worker;
+    this._version = version;
+    this._sampleRate = sampleRate;
+    this._maxCharacterLimit = maxCharacterLimit;
+    this._validPunctuationSymbols = validPunctuationSymbols;
+    this._numSymbols = numSymbols;
+  }
+
+  /**
+   * Get Orca engine version.
+   */
+  get version(): string {
+    return this._version;
+  }
+
+  /**
+   * Get sample rate.
+   */
+  get sampleRate(): number {
+    return this._sampleRate;
+  }
+
+  /**
+   * Get maximum character limit.
+   */
+  get maxCharacterLimit(): number {
+    return this._maxCharacterLimit;
```

Comment:
after fixing the memory issue, we should test this and make sure that Orca works fine given the size of wasm memory